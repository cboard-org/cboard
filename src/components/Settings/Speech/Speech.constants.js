export const MIN_PITCH = 0.0;
export const MAX_PITCH = 2.0;
export const INCREMENT_PITCH = 0.25;
export const MIN_RATE = 0;
export const MAX_RATE = 2;
export const INCREMENT_RATE = 0.25;
