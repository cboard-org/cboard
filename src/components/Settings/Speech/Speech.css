.Speech__ListItemText {
  max-width: 50%;
}

.Speech__VoiceMenuItemText {
  display: flex;
  flex-direction: row;
  align-items: center;
  min-width: 100%;
}

.Speech__VoiceLabel:hover {
  overflow: visible;
  white-space: normal;
  height: auto;
}

.Speech__VoiceLabel {
  overflow: hidden;
  text-overflow: ellipsis;
  margin-right: 20px;
  white-space: nowrap;
  max-width: 80%;
}

.MuiChip-root {
  margin-left: auto;
}
