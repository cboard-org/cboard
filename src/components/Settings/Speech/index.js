export { default } from './Speech.container';
