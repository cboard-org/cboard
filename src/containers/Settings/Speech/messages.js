import { defineMessages } from 'react-intl';

export default defineMessages({
  higher: {
    id: 'cboard.containers.Settings.Speech.higher',
    defaultMessage: 'Higher'
  },
  lower: {
    id: 'cboard.containers.Settings.Speech.lower',
    defaultMessage: 'Lower'
  },
  faster: {
    id: 'cboard.containers.Settings.Speech.faster',
    defaultMessage: 'Faster'
  },
  slower: {
    id: 'cboard.containers.Settings.Speech.slower',
    defaultMessage: 'Slower'
  },
  pitch: {
    id: 'cboard.containers.Settings.Speech.pitch',
    defaultMessage: 'Pitch'
  },
  pitchDescription: {
    id: 'cboard.containers.Settings.Speech.pitchDescription',
    defaultMessage: 'Make the voice use a higher or lower pitch'
  },
  rate: {
    id: 'cboard.containers.Settings.Speech.rate',
    defaultMessage: 'Rate'
  },
  rateDescription: {
    id: 'cboard.containers.Settings.Speech.rateDescription',
    defaultMessage: 'Make the voice speak faster or slower'
  },
  speech: {
    id: 'cboard.containers.Settings.Speech.speech',
    defaultMessage: 'Speech'
  },
  sampleSentence: {
    id: 'cboard.containers.Settings.Speech.sampleSentence',
    defaultMessage: 'Hi! This is my voice.'
  },
  voice: {
    id: 'cboard.containers.Settings.Speech.voice',
    defaultMessage: 'Voice'
  }
});
