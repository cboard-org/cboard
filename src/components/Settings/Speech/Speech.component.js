import React from 'react';
import { FormattedMessage } from 'react-intl';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import Slider from '@material-ui/lab/Slider';

import FullScreenDialog from '../../UI/FullScreenDialog';
import {
  MIN_PITCH,
  MAX_PITCH,
  INCREMENT_PITCH,
  MIN_RATE,
  MAX_RATE,
  INCREMENT_RATE
} from './Speech.constants';
import messages from './Speech.messages';

const styles = theme => ({
  container: {
    display: 'flex',
    position: 'relative',
    justifyContent: 'center',
    width: '100%'
  }
});

const Speech = ({
  anchorEl,
  classes,
  handleChangePitch,
  handleChangeRate,
  handleClickListItem,
  handleMenuItemClick,
  handleVoiceClose,
  intl,
  langVoices,
  onClose,
  pitch,
  rate,
  selectedVoiceIndex,
  voiceOpen,
  voiceURI
}) => (
  <div className="Speech">
    <FullScreenDialog
      open
      title={<FormattedMessage {...messages.speech} />}
      onClose={onClose}
    >
      <Paper>
        <List>
          <ListItem
            button
            divider
            aria-haspopup="true"
            aria-controls="voice-menu"
            aria-label="Voice"
            onClick={handleClickListItem}
          >
            <ListItemText
              primary={<FormattedMessage {...messages.voice} />}
              secondary={voiceURI}
            />
          </ListItem>
          <ListItem divider aria-label={intl.formatMessage(messages.pitch)}>
            <ListItemText
              primary={<FormattedMessage {...messages.pitch} />}
              secondary={<FormattedMessage {...messages.pitchDescription} />}
            />
            <div className={classes.container}>
              <Slider
                value={pitch}
                min={MIN_PITCH}
                max={MAX_PITCH}
                step={INCREMENT_PITCH}
                onChange={handleChangePitch}
              />
            </div>
          </ListItem>
          <ListItem aria-label={intl.formatMessage(messages.rate)}>
            <ListItemText
              primary={<FormattedMessage {...messages.rate} />}
              secondary={<FormattedMessage {...messages.rateDescription} />}
            />
            <div className={classes.container}>
              <Slider
                value={rate}
                min={MIN_RATE}
                max={MAX_RATE}
                step={INCREMENT_RATE}
                onChange={handleChangeRate}
              />
            </div>
          </ListItem>
        </List>
      </Paper>
      <Menu
        id="voice-menu"
        anchorEl={anchorEl}
        open={voiceOpen}
        onClose={handleVoiceClose}
      >
        {langVoices.map((voice, index) => (
          <MenuItem
            key={index}
            selected={index === selectedVoiceIndex}
            onClick={() => handleMenuItemClick(voice, index)}
          >
            {voice.name}
          </MenuItem>
        ))}
      </Menu>
    </FullScreenDialog>
  </div>
);

export default withStyles(styles)(Speech);
