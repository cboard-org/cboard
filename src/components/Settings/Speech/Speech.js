import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import compose from 'recompose/compose';
import debounce from 'lodash.debounce';
import { withStyles } from 'material-ui/styles';
import { injectIntl, FormattedMessage } from 'react-intl';
import Button from 'material-ui/Button';
import List, { ListItem, ListItemText } from 'material-ui/List';
import Menu, { MenuItem } from 'material-ui/Menu';
import { LinearProgress } from 'material-ui/Progress';
import ArrowDownwardIcon from 'material-ui-icons/ArrowDownward';
import ArrowUpwardIcon from 'material-ui-icons/ArrowUpward';
import FastForwardIcon from 'material-ui-icons/FastForward';
import FastRewindIcon from 'material-ui-icons/FastRewind';

import {
  speak,
  changeVoice,
  changePitch,
  changeRate
} from '../../SpeechProvider/SpeechProvider.actions';
import FullScreenDialog from '../../FullScreenDialog';
import messages from './Speech.messages';

const MIN_PITCH = 0.0;
const MAX_PITCH = 2.0;
const INCREMENT_PITCH = 0.25;
const MIN_RATE = 0.0;
const MAX_RATE = 2.0;
const INCREMENT_RATE = 0.25;

const styles = theme => ({
  container: {
    display: 'flex',
    position: 'relative',
    justifyContent: 'center'
  },
  icon: {
    marginLeft: 3,
    marginRight: 3,
    width: 20
  },
  progress: {
    paddingLeft: 2,
    paddingRight: 2,
    maxWidth: '50%',
    minWidth: '20%',
    marginTop: 20
  }
});

const getProgressPercent = (value, min, max) =>
  Math.round((value - min) / (max - min) * 100.0);

export class Speech extends PureComponent {
  static propTypes = {
    /**
     * If true, Speech will be visible
     */
    open: PropTypes.bool,
    /**
     * Active language
     */
    lang: PropTypes.string,
    speech: PropTypes.object,
    voices: PropTypes.array,
    onRequestClose: PropTypes.func,
    changeVoice: PropTypes.func,
    changePitch: PropTypes.func,
    changeRate: PropTypes.func
  };

  state = {
    selectedVoiceIndex: 0,
    voiceOpen: false,
    anchorEl: null
  };

  speakSample = debounce(
    () => {
      const { intl, speak } = this.props;
      const text = intl.formatMessage(messages.sampleSentence);
      speak(text);
    },
    500,
    {}
  );

  handleClickListItem = event => {
    this.setState({ voiceOpen: true, anchorEl: event.currentTarget });
  };

  handleMenuItemClick = ({ voiceURI, lang }) => {
    const { changeVoice } = this.props;
    changeVoice(voiceURI, lang);
    this.speakSample();
    this.setState({ voiceOpen: false });
  };

  handleChangePitch = value => {
    const { changePitch } = this.props;
    changePitch(value);
    this.speakSample();
  };

  handleChangeRate = value => {
    const { changeRate } = this.props;
    changeRate(value);
    this.speakSample();
  };

  handleVoiceRequestClose = () => {
    this.setState({ voiceOpen: false });
  };

  render() {
    const {
      intl,
      classes,
      open,
      lang,
      onRequestClose,
      speech: { voices, options: { voiceURI, pitch, rate } }
    } = this.props;

    const langVoices = voices.filter(
      voice => voice.lang.slice(0, 2) === lang.slice(0, 2)
    );

    return (
      <div className="Speech">
        <FullScreenDialog
          open={open}
          title={<FormattedMessage {...messages.speech} />}
          onRequestClose={onRequestClose}
        >
          <List>
            <ListItem
              button
              divider
              aria-haspopup="true"
              aria-controls="voice-menu"
              aria-label="Voice"
              onClick={this.handleClickListItem}
            >
              <ListItemText
                primary={<FormattedMessage {...messages.voice} />}
                secondary={voiceURI}
              />
            </ListItem>
            <ListItem divider aria-label={intl.formatMessage(messages.pitch)}>
              <ListItemText
                primary={<FormattedMessage {...messages.pitch} />}
                secondary={<FormattedMessage {...messages.pitchDescription} />}
              />
              <div className={classes.container}>
                <Button
                  color="primary"
                  aria-label={intl.formatMessage(messages.lower)}
                  disabled={pitch <= MIN_PITCH}
                  onClick={() =>
                    this.handleChangePitch(pitch - INCREMENT_PITCH)
                  }
                >
                  <ArrowDownwardIcon className={classes.icon} />
                </Button>
                <div className={classes.progress}>
                  <LinearProgress
                    mode="determinate"
                    value={getProgressPercent(pitch, MIN_PITCH, MAX_PITCH)}
                  />
                </div>
                <Button
                  color="primary"
                  aria-label={intl.formatMessage(messages.higher)}
                  disabled={pitch >= MAX_PITCH}
                  onClick={() =>
                    this.handleChangePitch(pitch + INCREMENT_PITCH)
                  }
                >
                  <ArrowUpwardIcon className={classes.icon} />
                </Button>
              </div>
            </ListItem>
            <ListItem aria-label={intl.formatMessage(messages.rate)}>
              <ListItemText
                primary={<FormattedMessage {...messages.rate} />}
                secondary={<FormattedMessage {...messages.rateDescription} />}
              />
              <div className={classes.container}>
                <Button
                  color="primary"
                  aria-label={intl.formatMessage(messages.slower)}
                  disabled={rate <= MIN_RATE}
                  onClick={() => this.handleChangeRate(rate - INCREMENT_RATE)}
                >
                  <FastRewindIcon className={classes.icon} />
                </Button>
                <div className={classes.progress}>
                  <LinearProgress
                    mode="determinate"
                    value={getProgressPercent(rate, MIN_RATE, MAX_RATE)}
                  />
                </div>
                <Button
                  color="primary"
                  aria-label={intl.formatMessage(messages.faster)}
                  disabled={rate >= MAX_RATE}
                  onClick={() => this.handleChangeRate(rate + INCREMENT_RATE)}
                >
                  <FastForwardIcon className={classes.icon} />
                </Button>
              </div>
            </ListItem>
          </List>
          <Menu
            id="voice-menu"
            anchorEl={this.state.anchorEl}
            open={this.state.voiceOpen}
            onRequestClose={this.handleVoiceRequestClose}
          >
            {langVoices.map((voice, index) => (
              <MenuItem
                key={index}
                selected={index === this.state.selectedVoiceIndex}
                onClick={event => this.handleMenuItemClick(voice)}
              >
                {voice.name}
              </MenuItem>
            ))}
          </Menu>
        </FullScreenDialog>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    lang: state.language.lang,
    voices: state.speech.voices,
    speech: state.speech
  };
};

export function mapDispatchToProps(dispatch) {
  return {
    changeVoice: (voiceURI, lang) => {
      dispatch(changeVoice(voiceURI, lang));
    },
    changePitch: pitch => {
      dispatch(changePitch(pitch));
    },
    changeRate: rate => {
      dispatch(changeRate(rate));
    },
    speak: text => {
      dispatch(speak(text));
    }
  };
}

const EnhancedSpeech = compose(injectIntl, withStyles(styles))(Speech);
export default connect(mapStateToProps, mapDispatchToProps)(EnhancedSpeech);
