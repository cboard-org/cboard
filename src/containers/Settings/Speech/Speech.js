import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import compose from 'recompose/compose';
import debounce from 'lodash.debounce';
import { withStyles } from 'material-ui/styles';
import { injectIntl, FormattedMessage } from 'react-intl';
import Button from 'material-ui/Button';
import List, { ListItem, ListItemText } from 'material-ui/List';
import Menu, { MenuItem } from 'material-ui/Menu';
import { LinearProgress } from 'material-ui/Progress';
import ArrowDownwardIcon from 'material-ui-icons/ArrowDownward';
import ArrowUpwardIcon from 'material-ui-icons/ArrowUpward';
import FastForwardIcon from 'material-ui-icons/FastForward';
import FastRewindIcon from 'material-ui-icons/FastRewind';

import { changeVoice, changePitch, changeRate } from '../../../speech/actions';
import speech from '../../../speech';
import FullScreenDialog from '../../../components/FullScreenDialog';
import messages from './messages';

const MIN_PITCH = 0.0;
const MAX_PITCH = 2.0;
const INCREMENT_PITCH = 0.25;
const MIN_RATE = 0.0;
const MAX_RATE = 2.0;
const INCREMENT_RATE = 0.25;

const styles = theme => ({
  container: {
    display: 'flex',
    position: 'relative',
    justifyContent: 'center'
  },
  icon: {
    marginLeft: 3,
    marginRight: 3,
    width: 20
  },
  progress: {
    paddingLeft: 2,
    paddingRight: 2,
    maxWidth: '50%',
    minWidth: '20%',
    marginTop: 20
  }
});

const getProgressPercent = (value, min, max) =>
  Math.round((value - min) / (max - min) * 100.0);

export class Speech extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      selectedVoiceIndex: 0,
      voiceOpen: false,
      anchorEl: null
    };
  }

  speakSample = debounce(
    () => {
      const { intl } = this.props;
      const text = intl.formatMessage(messages.sampleSentence);
      speech.speak(text);
    },
    500,
    {}
  );

  handleClickListItem = (event) => {
    this.setState({ voiceOpen: true, anchorEl: event.currentTarget });
  };

  handleMenuItemClick = ({ voiceURI, lang }) => {
    const { changeVoice } = this.props;
    changeVoice(voiceURI, lang);
    this.speakSample();
    this.setState({ voiceOpen: false });
  };

  handleChangePitch = value => {
    const { changePitch } = this.props;
    changePitch(value);
    this.speakSample();
  };

  handleChangeRate = value => {
    const { changeRate } = this.props;
    changeRate(value);
    this.speakSample();
  };

  handleVoiceRequestClose = () => {
    this.setState({ voiceOpen: false });
  };

  render() {
    const {
      open,
      locale,
      onCancel,
      speech: { voices, voiceURI, pitch, rate },
      classes,
      intl
    } = this.props;

    const localeVoices = voices.filter(
      voice => voice.lang.slice(0, 2) === locale
    );

    return (
      <div className="Speech">
        <FullScreenDialog
          open={open}
          title={<FormattedMessage {...messages.speech} />}
          onCancel={onCancel}
        >
          <List>
            <ListItem
              button
              divider
              aria-haspopup="true"
              aria-controls="voice-menu"
              aria-label="Voice"
              onClick={this.handleClickListItem}
            >
              <ListItemText
                primary={<FormattedMessage {...messages.voice} />}
                secondary={voiceURI}
              />
            </ListItem>
            <ListItem divider aria-label={intl.formatMessage(messages.pitch)}>
              <ListItemText
                primary={<FormattedMessage {...messages.pitch} />}
                secondary={<FormattedMessage {...messages.pitchDescription} />}
              />
              <div className={classes.container}>
                <Button
                  color="primary"
                  aria-label={intl.formatMessage(messages.lower)}
                  disabled={pitch <= MIN_PITCH}
                  onClick={() =>
                    this.handleChangePitch(pitch - INCREMENT_PITCH)}
                >
                  <ArrowDownwardIcon className={classes.icon} />
                </Button>
                <div className={classes.progress}>
                  <LinearProgress
                    mode="determinate"
                    value={getProgressPercent(pitch, MIN_PITCH, MAX_PITCH)}
                  />
                </div>
                <Button
                  color="primary"
                  aria-label={intl.formatMessage(messages.higher)}
                  disabled={pitch >= MAX_PITCH}
                  onClick={() =>
                    this.handleChangePitch(pitch + INCREMENT_PITCH)}
                >
                  <ArrowUpwardIcon className={classes.icon} />
                </Button>
              </div>
            </ListItem>
            <ListItem aria-label={intl.formatMessage(messages.rate)}>
              <ListItemText
                primary={<FormattedMessage {...messages.rate} />}
                secondary={<FormattedMessage {...messages.rateDescription} />}
              />
              <div className={classes.container}>
                <Button
                  color="primary"
                  aria-label={intl.formatMessage(messages.slower)}
                  disabled={rate <= MIN_RATE}
                  onClick={() => this.handleChangeRate(rate - INCREMENT_RATE)}
                >
                  <FastRewindIcon className={classes.icon} />
                </Button>
                <div className={classes.progress}>
                  <LinearProgress
                    mode="determinate"
                    value={getProgressPercent(rate, MIN_RATE, MAX_RATE)}
                  />
                </div>
                <Button
                  color="primary"
                  aria-label={intl.formatMessage(messages.faster)}
                  disabled={rate >= MAX_RATE}
                  onClick={() => this.handleChangeRate(rate + INCREMENT_RATE)}
                >
                  <FastForwardIcon className={classes.icon} />
                </Button>
              </div>
            </ListItem>
          </List>
          <Menu
            id="voice-menu"
            anchorEl={this.state.anchorEl}
            open={this.state.voiceOpen}
            onRequestClose={this.handleVoiceRequestClose}
          >
            {localeVoices.map((voice, index) =>
              <MenuItem
                key={index}
                selected={index === this.state.selectedVoiceIndex}
                onClick={event => this.handleMenuItemClick(voice)}
              >
                {voice.name}
              </MenuItem>
            )}
          </Menu>
        </FullScreenDialog>
      </div>
    );
  }
}

Speech.propTypes = {
  open: PropTypes.bool,
  locale: PropTypes.string,
  speech: PropTypes.object,
  voices: PropTypes.array,
  onCancel: PropTypes.func,
  changeVoice: PropTypes.func,
  changePitch: PropTypes.func,
  changeRate: PropTypes.func
};

const mapStateToProps = state => {
  return {
    locale: state.language.locale,
    voices: state.speech.voices,
    speech: state.speech
  };
};

export function mapDispatchToProps(dispatch) {
  return {
    changeVoice: (voiceURI, lang) => {
      dispatch(changeVoice(voiceURI, lang));
    },
    changePitch: pitch => {
      dispatch(changePitch(pitch));
    },
    changeRate: rate => {
      dispatch(changeRate(rate));
    }
  };
}

const EnhancedSpeech = compose(injectIntl, withStyles(styles))(Speech);
export default connect(mapStateToProps, mapDispatchToProps)(EnhancedSpeech);
