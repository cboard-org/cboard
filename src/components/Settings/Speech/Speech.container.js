import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import debounce from 'lodash.debounce';
import { injectIntl } from 'react-intl';

import {
  speak,
  changeVoice,
  changePitch,
  changeRate
} from '../../../providers/SpeechProvider/SpeechProvider.actions';
import Speech from './Speech.component';
import messages from './Speech.messages';

export class SpeechContainer extends Component {
  static propTypes = {
    /**
     * Active language
     */
    lang: PropTypes.string,
    speech: PropTypes.object,
    voices: PropTypes.array,
    changeVoice: PropTypes.func,
    changePitch: PropTypes.func,
    changeRate: PropTypes.func,
    history: PropTypes.object.isRequired
  };

  state = {
    selectedVoiceIndex: 0,
    voiceOpen: false,
    anchorEl: null
  };

  speakSample = debounce(() => {
    const { intl, speak } = this.props;

    const text = intl.formatMessage(messages.sampleSentence);
    speak(text);
  }, 500);

  handleClickListItem = event => {
    this.setState({ voiceOpen: true, anchorEl: event.currentTarget });
  };

  handleMenuItemClick = ({ voiceURI, lang }, index) => {
    const { changeVoice } = this.props;

    changeVoice(voiceURI, lang);
    this.speakSample();
    this.setState({ voiceOpen: false, selectedVoiceIndex: index });
  };

  handleChangePitch = value => {
    const { changePitch } = this.props;

    changePitch(value);
    this.speakSample();
  };

  handleChangeRate = value => {
    const { changeRate } = this.props;

    changeRate(value);
    this.speakSample();
  };

  handleVoiceRequestClose = () => {
    this.setState({ voiceOpen: false });
  };

  render() {
    const {
      history,
      intl,
      lang,
      speech: { voices, options: { voiceURI, pitch, rate } }
    } = this.props;

    const langVoices = voices.filter(
      voice => voice.lang.slice(0, 2) === lang.slice(0, 2)
    );

    return (
      <Speech
        {...this.state}
        handleChangePitch={this.handleChangePitch}
        handleChangeRate={this.handleChangeRate}
        handleClickListItem={this.handleClickListItem}
        handleMenuItemClick={this.handleMenuItemClick}
        handleVoiceRequestClose={this.handleVoiceRequestClose}
        intl={intl}
        langVoices={langVoices}
        onRequestClose={history.goBack}
        pitch={pitch}
        rate={rate}
        voiceURI={voiceURI}
      />
    );
  }
}

const mapStateToProps = state => ({
  lang: state.language.lang,
  voices: state.speech.voices,
  speech: state.speech
});

const mapDispatchToProps = {
  changeVoice,
  changePitch,
  changeRate,
  speak
};

const EnhancedSpeech = injectIntl(Speech);

export default connect(mapStateToProps, mapDispatchToProps)(EnhancedSpeech);
