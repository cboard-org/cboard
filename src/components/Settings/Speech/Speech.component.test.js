import React from 'react';
import { shallowMatchSnapshot } from '../../../common/test_utils';
import Speech from './Speech.component';

jest.mock('./Speech.messages', () => {
  return {
    higher: {
      id: 'cboard.components.Settings.Speech.higher',
      defaultMessage: 'Higher'
    },
    lower: {
      id: 'cboard.components.Settings.Speech.lower',
      defaultMessage: 'Lower'
    },
    faster: {
      id: 'cboard.components.Settings.Speech.faster',
      defaultMessage: 'Faster'
    },
    slower: {
      id: 'cboard.components.Settings.Speech.slower',
      defaultMessage: 'Slower'
    },
    pitch: {
      id: 'cboard.components.Settings.Speech.pitch',
      defaultMessage: 'Pitch'
    },
    pitchDescription: {
      id: 'cboard.components.Settings.Speech.pitchDescription',
      defaultMessage: 'Make the voice use a higher or lower pitch'
    },
    rate: {
      id: 'cboard.components.Settings.Speech.rate',
      defaultMessage: 'Rate'
    },
    rateDescription: {
      id: 'cboard.components.Settings.Speech.rateDescription',
      defaultMessage: 'Make the voice speak faster or slower'
    },
    speech: {
      id: 'cboard.components.Settings.Speech.speech',
      defaultMessage: 'Speech'
    },
    sampleSentence: {
      id: 'cboard.components.Settings.Speech.sampleSentence',
      defaultMessage: 'Hi! This is my voice.'
    },
    voice: {
      id: 'cboard.components.Settings.Speech.voice',
      defaultMessage: 'Voice'
    }
  };
});

describe('Speech tests', () => {
  test('default renderer', () => {
    shallowMatchSnapshot(<Speech />);
  });
});
