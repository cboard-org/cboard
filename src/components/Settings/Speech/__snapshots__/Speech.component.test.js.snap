// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Speech tests default renderer 1`] = `
<Speech
  classes={
    Object {
      "container": "Speech-container-1",
    }
  }
/>
`;
